File: examples/relay-cascade/cascade-node-cpp/api/public.cpp
#include "api/public.hpp"

#include "holons/observability.hpp"

#include <filesystem>

namespace cascade::node::cppholon::api {
namespace {

std::string ResponderSlug(const holons::observability::Observability &obs) {
  if (!obs.cfg.slug.empty()) {
    return obs.cfg.slug;
  }
  return "cascade-node-cpp";
}

} // namespace

::relay::v1::TickResponse Tick(const ::relay::v1::TickRequest &) {
  auto &obs = holons::observability::current();
  ::relay::v1::TickResponse response;
  response.set_responder_slug(ResponderSlug(obs));
  response.set_responder_instance_uid(obs.cfg.instance_uid);
  return response;
}

} // namespace cascade::node::cppholon::api

File: examples/observability-cascade/observability-cascade-go-node/api/public.go
package api

import (
	"context"

	pb "observability-cascade-go-node/gen/go/relay/v1"
	"observability-cascade-go-node/internal"
)

// Tick emits one cascade tick through the in-process implementation.
func Tick(req *pb.TickRequest) (*pb.TickResponse, error) {
	return (&internal.Server{}).Tick(context.Background(), req)
}
